src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -914,10 +914,21 @@ private static bool IsPropertyTrue(IMSBuildItem item, string propertyName)
         /// <param name="logger">Logger used to display warnings and errors.</param>
         public static Task ReplayWarningsAndErrorsAsync(LockFile lockFile, ILogger logger)
         {
-            var logMessages = lockFile?.LogMessages?.Select(m => m.AsRestoreLogMessage()) ??
-                Enumerable.Empty<RestoreLogMessage>();
+            return ReplayWarningsAndErrorsAsync(lockFile?.LogMessages, logger);
+        }
+
+        /// <summary>
+        /// Function used to display errors and warnings at the end of restore operation.
+        /// The errors and warnings are read from the assets file based on restore result.
+        /// </summary>
+        /// <param name="messages">Messages to log.</param>
+        /// <param name="logger">Logger used to display warnings and errors.</param>
+        public static Task ReplayWarningsAndErrorsAsync(IList<IAssetsLogMessage> messages, ILogger logger)
```

Comment:
Add unit tests for this.  Should be easy since everything is an interface.

Reply:
There's already a test, `MSBuildRestoreUtility_ReplayWarningsAndErrors_Minimal`.  Do you want something different?